packages/application/src/index.ts

```diff
@@ -299,21 +398,22 @@ export class Application<T extends Widget> {
    * the required services for the user's plugins will be resolved
    * automatically when the plugin is activated.
    */
-  resolveRequiredService<U>(token: Token<U>): Promise<U> {
+  async resolveRequiredService<U>(token: Token<U>): Promise<U> {
     // Reject the promise if there is no provider for the type.
     let id = this._serviceMap.get(token);
     if (!id) {
       return Promise.reject(new Error(`No provider for: ${token.name}.`));
```

Comment:
We can just `throw` since you've converted it to `async`

Reply:
I use my editor refactoring tool to switch to `async-await` as I find that syntax easier to read - especially for people unfamiliar with Promises. So I did not look too much to those - thanks for doing it.